java2typescript-jackson/src/test/resources/java2typescript/jackson/module/WriterPreferencesTest.enumToStringLiteralType.d.ts

```diff
@@ -3,5 +3,11 @@ export type Enum =
     | "VAL2"
     | "name";
 
+export interface TestClassHasMapWithEnumKey {
+    enumKeyMap: { [key in Enum ]: string;};
```

Comment:
There is one thing, that i'm not quite happy with this approach (of course still much better than the bug You encountered...)

Looks like this means that `enumKeyMap` must contain all keys of the Enum, if some are missing, there is a compile-time error. This doesn't act like the original Java field `Map<Enum, String> enumKeyMap` where You don't have to provide all enum keys to initialize the variable/field. So i think I'd feel uncomfortable with this solution.

I hoped that perhaps TypeScript 2.1 feature `Partial<T>` could be used to specify, that it is not required to provide all enum constants:
```TypeScript
...
export class EnumValues { // generated when `moduleWriter.preferences.useStringLiteralTypeForEnums(true);`
    static VAL1: Enum = "VAL1";
    static VAL2: Enum = "VAL2";
    ...
}

type PartialEnumValues = Partial<EnumValues>; // FIXME: type is `{}`, not `{VAL1?: Enum, VAL2?: Enum, ...}` :(

export interface TestClassHasMapWithEnumKey {
    enumKeyMap: { [key in keyof PartialEnumValues]: string; }; // useless - any key would be accepted
    ...
 }
```
but this doesn't work as i first expected either, because `Partial<EnumValues>` doesn't take static fields from the EnumValues class. So I'd say that for this use-case it would be nice, if `PartialEnumValues` would be also generated next to enum (at least when it is already used in index signature in the generated code, but it could be useful for manually written code as well). Then the compiler would allow to use only existing enum constants as the keys and wouldn't require You to set all of them.